django/apps/existing_database/models.py

```diff
@@ -0,0 +1,213 @@
+    user_group_id = models.CharField(primary_key=True, max_length=-1)
+    name = models.CharField(max_length=-1, null=True)
+    description = models.TextField(blank=True, null=True)
+    created_at = models.DateTimeField(blank=True, null=True)
+    created_by = models.ForeignKey(User, models.DO_NOTHING)
+    archived_at = models.DateTimeField(blank=True, null=True)
+    archived_by = models.ForeignKey(User, models.DO_NOTHING)
+    is_archived = models.BooleanField(blank=True, null=True)
+
+    class Meta:
+        managed = False
+        db_table = "user_groups"
+        app_label = settings.MAPSWIPE_EXISTING_DB
+
+    def __str__(self):
+        return self.user_group_id
+
+    def user_memberships(self):
+        return UserGroupUserMembership.objects.filter(
+            user_group_id=self.user_group_id
+        ).select_related("user")
+
+
+class UserGroupUserMembership(models.Model):
+    user_group = models.ForeignKey(UserGroup, models.DO_NOTHING, primary_key=True)
+    user = models.ForeignKey(User, models.DO_NOTHING, primary_key=True)
+    is_active = models.BooleanField(null=True, blank=True)
+    left_at = ArrayField(models.DateTimeField(blank=True, null=True), default=dict)
+    joined_at = ArrayField(models.DateTimeField(blank=True, null=True), default=dict)
+
+    class Meta:
+        managed = False
+        db_table = "user_groups_user_memberships"
+        unique_together = (("user_group", "user"),)
+        app_label = settings.MAPSWIPE_EXISTING_DB
+
+    def __str__(self):
+        return f"UG:{self.user_group_id}-U:{self.user_id}"
+
+
+class Organization(models.Model):
+    organization_id = models.CharField(primary_key=True, max_length=-1)
+
+    class Meta:
+        managed = False
+        db_table = "organizations"
+        app_label = settings.MAPSWIPE_EXISTING_DB
+
+    def __str__(self):
+        return self.organization_id
+
+
+class Project(models.Model):
+    project_id = models.CharField(primary_key=True, max_length=-1)
+    created = models.DateTimeField(blank=True, null=True)
+    created_by = models.CharField(max_length=-1, blank=True, null=True)
+    geom = gis_models.GeometryField(blank=True, null=True)
+    image = models.CharField(max_length=-1, blank=True, null=True)
+    is_featured = models.BooleanField(blank=True, null=True)
+    look_for = models.CharField(max_length=-1, blank=True, null=True)
+    name = models.CharField(max_length=-1, blank=True, null=True)
+    progress = models.IntegerField(blank=True, null=True)
+    project_details = models.CharField(max_length=-1, blank=True, null=True)
+    project_type = models.IntegerField(blank=True, null=True)
+    required_results = models.IntegerField(blank=True, null=True)
+    result_count = models.IntegerField(blank=True, null=True)
+    status = models.CharField(max_length=-1, blank=True, null=True)
+    verification_number = models.IntegerField(blank=True, null=True)
+    # Database uses JSON instead of JSONB (not supported by django)
```

Comment:
should we use JSONB then?

Reply:
I will recommend using JSONB. But maybe we can discuss this later.

> And this has some immediate benefits:
> - More efficiency.
> - Significantly faster to process.
> - Supports indexing (which can be a significant advantage, as we’ll see later). simpler schema designs (replacing entity-attribute-value (EAV) tables with JSONB columns, which can be queried, indexed, and joined, allowing for performance improvements up until 1000X!)
> 
> Few Drawbacks:
> - Slightly slower input (due to added conversion overhead).
> - It may take more disk space than plain JSON due to a larger table footprint, though not always.

Reply:
@Hagellach37 should we create a ticket for these kind of improvements?